pki/ejbca/api-docs/apiary.apib

```diff
@@ -0,0 +1,490 @@
+          "moreinfo": "https://dojot.github.io/ejbca-rest/apiary_latest.html"
+        }
+
++ Response 200 (application/json)
+
+        {
+          "CRL": "MIICQjCCASoCAQEwDQYJKoZIhvcNAQEFBQAwEzERMA8GA1UEAwwIUF..."
+        }
+
+# Group EJBCA user management
+## Create or edit user [/user]
+### Create or edit a user [POST]
+
++ Request (application/json)
+  + Body
+
+            {
+                "username" : "device21"
+            }
+
++ Response 400 (application/json)
+
+        {
+          "code": 400,
+          "message": "username must have more than 3 characters.",
+          "moreinfo": "https://dojot.github.io/ejbca-rest/apiary_latest.html"
+        }
+
++ Response 500 (application/json)
+
+        {
+          "code": 500,
+          "message": "Internal error.",
+          "moreinfo": "https://dojot.github.io/ejbca-rest/apiary_latest.html"
+        }
+
++ Response 200 (application/json)
+
+        {
+          "status": 200,
+          "message": "user created/edited with success."
+        }
+
+## User CRUD [/user/{username}]
+### get user data [GET]
+
++ Parameters
+  + username: IOTmidCA (required, string): user username
+
+
++ Response 400 (application/json)
+
+        {
+          "code": 400,
+          "message": "soap message: query exception.",
+          "moreinfo": "https://dojot.github.io/ejbca-rest/apiary_latest.html"
+        }
+
++ Response 404 (application/json)
+
+        {
+          "code": 404,
+          "message": "User does not exist.",
+          "moreinfo": "https://dojot.github.io/ejbca-rest/apiary_latest.html"
+        }
+
++ Response 200 (application/json)
+
+        {
+          "user": [
+              {
+                  "caName": "PROTOCOL",
+                  "cardNumber": null,
+                  "certificateProfileName": "ENDUSER",
+                  "certificateSerialNumber": null,
+                  "clearPwd": false,
+                  "email": "device21@noemail.com",
+                  "endEntityProfileName": "EMPTY",
+                  "endTime": null,
+                  "extendedInformation": [
+                      {
+                          "name": "subjectdirattributes",
+                          "value": null
+                      },
+                      {
+                          "name": "somekey",
+                          "value": "somevalue"
+                      }
+                  ],
+                  "hardTokenIssuerName": null,
+                  "keyRecoverable": false,
+                  "password": null,
+                  "sendNotification": false,
+                  "startTime": null,
+                  "status": 40,
+                  "subjectAltName": null,
+                  "subjectDN": "CN=device21",
+                  "tokenType": "USERGENERATED",
+                  "username": "device21"
+              }
+          ]
+        }
+
+
+### Revoke all cerificates from a user [DELETE /user/{username}?{reason,delete}]
```

Comment:
This endpoint doesn't follow the good practices.

DELETE /user/{username} - remove the resource (user)
POST /user/{username}/revoke_all_certificates - revoke all certificates